h2quic/response_writer.go

```diff
@@ -21,18 +25,26 @@ type responseWriter struct {
 	headerStream      quic.Stream
 	headerStreamMutex *sync.Mutex
 
+	// Server Push
+	session     streamCreator
+	handlerFunc http.HandlerFunc
+	requestHost string
+
 	header        http.Header
 	status        int // status code passed to WriteHeader
 	headerWritten bool
 }
 
-func newResponseWriter(headerStream quic.Stream, headerStreamMutex *sync.Mutex, dataStream quic.Stream, dataStreamID protocol.StreamID) *responseWriter {
+func newResponseWriter(headerStream quic.Stream, headerStreamMutex *sync.Mutex, dataStream quic.Stream, dataStreamID protocol.StreamID, session streamCreator, handlerFunc http.HandlerFunc, requestHost string) *responseWriter {
```

Comment:
This is getting a bit complicated. Somehow I have the feeling that the `responseWriter` isn't a good abstraction anymore. There's nothing you need to do here at this point, for now I'm fine with merging this as is, but we'll probably need to refactor things at some point.